Builder/Admin/ListBuilder.php

```diff
@@ -49,7 +49,11 @@ public function getFilterColumns()
 
     protected function findFilterColumns()
     {
-        foreach ($this->getAllFields() as $columnName) {
+        $columns = $this->getVariable('filters');
+        if (null == $columns || 0 == sizeof($columns)) {
+            $columns = $this->getDisplayColumns();
```

Comment:
Finally not.

I'm working on this, and, I'm not sure this is enough.

I should push a PR including this change and some more. I think will can close that one